2023/articles/2023-12-23.pod

```diff
@@ -0,0 +1,295 @@
+Author: olaf@wundersolutions.com
+Title: Elves Versus Unused Imports
+Topic: App::perlimports
+
+=encoding utf8
+
+Santa's elves are interested in code quality. This is something we touched on
+about 9 years ago in L<How Santa's Elves Keep their Workshop
+Tidy|https://perladvent.org/2014/2014-12-16.html> and, more recently, in
+L<Elves Versus Typos|https://perladvent.org/2023/2023-12-21.html>. Since the
+elves like keeping up with the times, they are currently experimenting with
+another, newer Perl code quality tool. It's finally time to take a close look
+at
+L<perlimports|https://metacpan.org/dist/App-perlimports/view/script/perlimports>.
+L<perlimports|https://metacpan.org/dist/App-perlimports/view/script/perlimports>
+is a fixer -- it can rewrite your code for you. It's also a linter, so you can
+use it to report on problems without fixing them. With this power comes great
+responsibility and C<perlimports> tries to be responsible. Inspired by
+L<goimports|https://pkg.go.dev/golang.org/x/tools/cmd/goimports>,
+C<perlimports> is an opinionated tool which tries to force its opinions on your
+code. What you get in return is a tool which takes much of the burden of
+managing imports out of your hands.
+
+If you're interested in an in-depth discussion of how C<perlimports> and Perl's
+C<use> and C<require> work, check out L<perlimports or "Where did that symbol come
+from?"|https://www.youtube.com/watch?v=fKqxdTbGxYY>.
+
+Let's take a look at a typical getting started workflow. First the elves
+install the package from CPAN:
+
+  cpm install -g App::perlimports
+
+Once that is done, the elves are ready to try this out. They'll probably just
+dive right in. The quick way to run C<perlimports> on a new repository might
+look something like this:
+
+=for :list
+* Clone a repo
+* Install *all* of the repository's dependencies (including recommended)
+* Run the test suite
+* Ensure all of the tests are passing. If there are test failures, fix those first
+* Run C<perlimports> with the C<--lint> flag to see what changes it might make
+* Tweak the configuration until we're happy with the linting results
+* Apply C<perlimports> to the tests. We can do this via C<perlimports -i t>
+* Ensure all of the tests are still passing
+* Commit the changes
+* Move on to other parts of the code, like C<lib>. C<perlimports --lint lib>
+
+In a best case scenario, this "just works". Let's try it on a really old
+repository of mine.
+
+    #!vim text
+    $ git clone https://github.com/oalders/acme-odometer.git
+    $ cd acme-odometer/
+    $ cpm install -g --with-recommends --cpanfile cpanfile
+    $ yath t
+    $ perlimports --lint t
+
+And indeed, all of the steps "just worked". We ran the test(s) via L<yath> and
```

Comment:
Any particular reason why you used yath here rather than `make test`? Might be nice to tell the readers why, even if it's just "brevity". Sometimes I find these little incidental points in the articles turn out to be very useful/interesting. :-)